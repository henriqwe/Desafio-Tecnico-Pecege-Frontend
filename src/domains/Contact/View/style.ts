import styled from 'styled-components'
import * as Icons from '@heroicons/react/24/outline'

export const Container = styled.section`
  position: relative;
  max-height: 100vh;
  min-height: 100vh;
  display: flex;
  flex: 1;
  flex-direction: column;
  overflow-x: none;
`

export const ChevronLeftIcon = styled(Icons.ChevronLeftIcon)`
  width: 2rem;
  height: 2rem;
`

export const EllipsisVerticalIcon = styled(Icons.EllipsisVerticalIcon)`
  width: 2rem;
  height: 2rem;
`

export const ContactHeader = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`
export const AvatarWrapper = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  padding-top: 1rem;
  gap: 0.4rem;
`
export const AvatarName = styled.p`
  font-size: 1.1rem;
  color: rgb(15 23 42);
  font-weight: 800;
`
export const AvatarPhone = styled.p`
  font-size: 0.9rem;
  color: rgb(100 116 139);
  font-weight: 400;
`
export const ContactHeaderWrapper = styled.div`
  background-color: rgba(100, 116, 139, 0.1);
  padding: 0.8rem;
  padding-top: 1rem;
`
export const UserIcon = styled(Icons.UserIcon)`
  width: 1.5rem;
  height: 1.5rem;
`
export const MapPinIcon = styled(Icons.MapPinIcon)`
  width: 1.5rem;
  height: 1.5rem;
`
export const BuildingStorefrontIcon = styled(Icons.BuildingStorefrontIcon)`
  width: 1.5rem;
  height: 1.5rem;
`
export const InfoTitle = styled.p`
  font-size: 0.9rem;
  color: rgb(15 23 42);
  font-weight: 800;
`
export const InfoDescription = styled.p`
  font-size: 1.1rem;
  color: rgb(100 116 139);
  font-weight: 400;
  padding-left: 1rem;
`
export const TabItemContainer = styled.section`
  display: flex;
  flex: 1;
  flex-direction: column;
  margin: 0.8rem;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.25);
`
export const InfoWrapper = styled.div<{ variant: 'white' | 'gray' }>`
  display: flex;
  flex: 1;
  gap: 0.2rem;
  padding: 0.7rem;
  flex-direction: column;
  background-color: ${({ variant }) =>
    variant === 'white' ? '#fff' : '#F1F5F9'};
`
export const DropdownOptionContent = styled.div`
  display: flex;
  gap: 0.5rem;
  align-items: end;
  justify-content: space-between;
  width: 100%;
`
export const PencilIcon = styled(Icons.PencilIcon)`
  width: 1.3rem;
  height: 1.3rem;
`
export const TrashIcon = styled(Icons.TrashIcon)`
  width: 1.3rem;
  height: 1.3rem;
  color: red;
`
export const MapContainer = styled.div`
  width: 100%;
  height: 20rem;
`
