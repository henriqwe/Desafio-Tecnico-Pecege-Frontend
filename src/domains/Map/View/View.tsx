import { FloatingActionButton } from '&components/FloatingActionButton/FloatingActionButton'
import { Map } from '&components/Map/Map'
import type { IContact } from '&types/contact'
import { fetchContacts } from '&services/queries/fetchContacts'
import { useQuery, useQueryClient } from '@tanstack/react-query'

export function MapView() {
  const queryClient = useQueryClient()
  async function handleFetch() {
    const contacts = queryClient.getQueryData(['contacts']) as IContact[]

    if (contacts) {
      return contacts
    }
    return fetchContacts()
  }
  const contactsQuery = useQuery({
    queryKey: ['contacts'],
    queryFn: () => handleFetch(),
    refetchOnWindowFocus: false,
    retry: false,
    suspense: true
  })

  return (
    <div style={{ width: '100vw', height: '100vh' }}>
      {!contactsQuery.data ? (
        <div></div>
      ) : (
        // <Loading />
        <>
          <Map contacts={contactsQuery.data} />
          <FloatingActionButton />
        </>
      )}
    </div>
  )
}
