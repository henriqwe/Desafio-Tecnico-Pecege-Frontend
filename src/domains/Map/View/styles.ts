import styled from 'styled-components'

export const MapViewContainer = styled.div`
  width: 100vw;
  height: 100vh;
`
