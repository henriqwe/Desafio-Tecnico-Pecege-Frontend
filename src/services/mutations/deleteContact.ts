import { fakePromise } from '&utils/fakePromise'

export async function deleteContact(id: number): Promise<number> {
  await fakePromise()

  return id
}
