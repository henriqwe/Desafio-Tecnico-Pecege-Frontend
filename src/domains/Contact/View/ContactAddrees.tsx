import * as S from './style'
import { IContact } from '&operations/queries/fetchContacts'
import { Map } from '&components/Map/Map'

export function ContactAddrees({ contact }: { contact: IContact }) {
  return (
    <S.TabItemContainer>
      <S.InfoWrapper variant="white">
        <S.InfoTitle>Rua</S.InfoTitle>
        <S.InfoDescription>{contact.address.street || '-'}</S.InfoDescription>
      </S.InfoWrapper>
      <S.InfoWrapper variant="gray">
        <S.InfoTitle>NÃºmero</S.InfoTitle>
        <S.InfoDescription>{contact.address.suite || '-'}</S.InfoDescription>
      </S.InfoWrapper>
      <S.InfoWrapper variant="white">
        <S.InfoTitle>Cidade</S.InfoTitle>
        <S.InfoDescription>{contact.address.city || '-'}</S.InfoDescription>
      </S.InfoWrapper>
      <S.InfoWrapper variant="gray">
        <S.InfoTitle>CEP</S.InfoTitle>
        <S.InfoDescription>{contact.address.zipcode || '-'}</S.InfoDescription>
      </S.InfoWrapper>
      {contact.address.geo.lat && contact.address.geo.lng && (
        <S.MapContainer>
          <Map contacts={[contact]} showInfoWindow={false} />
        </S.MapContainer>
      )}
    </S.TabItemContainer>
  )
}
