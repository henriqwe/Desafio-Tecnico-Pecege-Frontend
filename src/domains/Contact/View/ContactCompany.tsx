import * as S from './style'
import { IContact } from '&operations/queries/fetchContacts'

export function ContactCompany({ contact }: { contact: IContact }) {
  return (
    <S.TabItemContainer>
      <S.InfoWrapper variant="white">
        <S.InfoTitle>Nome</S.InfoTitle>
        <S.InfoDescription>{contact.company.name || '-'}</S.InfoDescription>
      </S.InfoWrapper>
      <S.InfoWrapper variant="gray">
        <S.InfoTitle>Frase de efeito</S.InfoTitle>
        <S.InfoDescription>
          {contact.company.catchPhrase || '-'}
        </S.InfoDescription>
      </S.InfoWrapper>
      <S.InfoWrapper variant="white">
        <S.InfoTitle>Estratégia de negócio</S.InfoTitle>
        <S.InfoDescription>{contact.company.bs || '-'}</S.InfoDescription>
      </S.InfoWrapper>
    </S.TabItemContainer>
  )
}
