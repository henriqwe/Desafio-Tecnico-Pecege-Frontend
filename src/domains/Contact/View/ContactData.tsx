import * as S from './style'
import type { IContact } from '&types/contact'

export function ContactData({ contact }: { contact: IContact }) {
  return (
    <S.TabItemContainer>
      <S.InfoWrapper variant="white">
        <S.InfoTitle>Nome</S.InfoTitle>
        <S.InfoDescription>{contact.name || '-'}</S.InfoDescription>
      </S.InfoWrapper>
      <S.InfoWrapper variant="gray">
        <S.InfoTitle>Telefone</S.InfoTitle>
        <S.InfoDescription>{contact.phone || '-'}</S.InfoDescription>
      </S.InfoWrapper>
      <S.InfoWrapper variant="white">
        <S.InfoTitle>E-mail</S.InfoTitle>
        <S.InfoDescription>{contact.email || '-'}</S.InfoDescription>
      </S.InfoWrapper>
      <S.InfoWrapper variant="gray">
        <S.InfoTitle>Website</S.InfoTitle>
        <S.InfoDescription>{contact.website || '-'}</S.InfoDescription>
      </S.InfoWrapper>
    </S.TabItemContainer>
  )
}
