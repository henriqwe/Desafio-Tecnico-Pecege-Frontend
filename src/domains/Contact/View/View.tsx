import { useNavigate, useParams } from 'react-router-dom'
import { Loading } from '&components/Loading/Loading'
import { routes } from '&utils/routes'
import { Avatar } from '&components/Avatar/Avatar'
import { useQueryClient, useMutation } from '@tanstack/react-query'
import { notification } from '&utils/notification'
import * as S from './style'
import { Tab, TabItem } from '&components/Tab/Tabs'
import { useState } from 'react'
import { ContactData } from './ContactData'
import { ContactAddrees } from './ContactAddrees'
import { ContactCompany } from './ContactCompany'
import { Dropdown } from '&components/Dropdown/Dropdown'
import { Modal } from '&components/Modal/Modal'
import { fakePromise } from '&utils/fakePromise'
import { useContact } from '&hooks/useContact'

type TSelectedTab = 0 | 1 | 2

export function ViewContact() {
  const { id } = useParams()

  const contact = useContact(id as string)

  const navigate = useNavigate()

  const onTabSelected = (index: number) => {
    setSelectedTab(index as TSelectedTab)
  }

  const [selectedTab, setSelectedTab] = useState<TSelectedTab>(0)
  const [openModal, setOpenModal] = useState(false)
  const queryClient = useQueryClient()

  const removeContact = useMutation({
    mutationFn: (id: number) => fakePromise(),
    onSuccess: (result, id) => {
      navigate(routes.home.path)
      queryClient.setQueryData(['contacts'], (old: any) =>
        old.filter((contact: any) => contact.id !== id)
      )

      notification('Contato removido com sucesso', 'success')
    },
    onError: (error: Error) => notification(error.message, 'error')
  })
  if (contact.error) {
    navigate(routes.home.path)
    notification(contact.error.message, 'error')
    return <></>
  }

  const TabsContent = {
    0: <ContactData contact={contact.data!} />,
    1: <ContactAddrees contact={contact.data!} />,
    2: <ContactCompany contact={contact.data!} />
  }
  const dropdownOption = [
    {
      content: (
        <S.DropdownOptionContent>
          Editar
          <S.PencilIcon />
        </S.DropdownOptionContent>
      ),
      fn: () => navigate(routes.editContact.path(contact.data?.id!))
    },
    {
      content: (
        <S.DropdownOptionContent>
          Remover
          <S.TrashIcon />
        </S.DropdownOptionContent>
      ),
      fn: () => setOpenModal(true)
    }
  ]
  return (
    <S.Container>
      {contact.isLoading || !contact.data ? (
        <S.LoadingContainer>
          <Loading />
        </S.LoadingContainer>
      ) : (
        <>
          <S.ContactHeaderWrapper>
            <S.ContactHeader>
              <div onClick={() => navigate(routes.home.path)}>
                <S.ChevronLeftIcon />
              </div>
              <h1>Contato</h1>

              <Dropdown options={dropdownOption} />
            </S.ContactHeader>

            <Modal
              action={() => removeContact.mutate(contact.data?.id!)}
              title="Deseja realmente remover esse contato?"
              actionsText="Sim"
              cancelText="Não"
              open={openModal}
              onOpenChange={setOpenModal}
            />
            <S.AvatarWrapper>
              <Avatar name={contact.data.name} variant="md" />
              <S.AvatarName>{contact.data.name}</S.AvatarName>
              <S.AvatarPhone>{contact.data.phone}</S.AvatarPhone>
            </S.AvatarWrapper>
          </S.ContactHeaderWrapper>
          <div>
            <Tab onTabSelected={onTabSelected}>
              <TabItem>
                <S.UserIcon />
                Dados
              </TabItem>
              <TabItem>
                <S.MapPinIcon />
                Endereço
              </TabItem>
              <TabItem>
                <S.BuildingStorefrontIcon />
                Empresa
              </TabItem>
            </Tab>
            <div>{TabsContent[selectedTab]}</div>
          </div>
        </>
      )}
    </S.Container>
  )
}
