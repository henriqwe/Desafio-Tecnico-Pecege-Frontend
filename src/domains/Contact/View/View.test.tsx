import { fireEvent, render, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'
import { TestProviders } from '&utils/tests/TestProviders'
import { ViewContact } from './View'
import * as toaster from '&utils/notification'
import userEvent from '@testing-library/user-event'
import * as mutationDeleteContact from '&services/mutations/deleteContact'
import { routes } from '&utils/routes'

const notification = jest.spyOn(toaster, 'notification')
const deleteContact = jest.spyOn(mutationDeleteContact, 'deleteContact')
const mockedUsedNavigate = jest.fn()

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate
}))
jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useParams: () => ({
    id: 1
  })
}))

describe('ViewContact', () => {
  it('should render the contact list correctly', async () => {
    const { container, findByTestId } = render(
      <TestProviders>
        <ViewContact />
      </TestProviders>
    )
    const tabEmpresa = await findByTestId('tab-empresa')
    const tabEndereco = await findByTestId('tab-endereco')

    await waitFor(async () => fireEvent.click(tabEmpresa))
    await waitFor(async () => fireEvent.click(tabEndereco))

    expect(container).toMatchSnapshot()
  })
  it('should go to edit contact page', async () => {
    const { findByTestId } = render(
      <TestProviders>
        <ViewContact />
      </TestProviders>
    )
    await waitFor(async () => findByTestId('avatar-container'))

    const dropdownMenuTrigger = await findByTestId('dropdown-menu-trigger')

    waitFor(() => userEvent.click(dropdownMenuTrigger))

    await waitFor(async () => findByTestId('edit-buttom'))

    const editButtom = await findByTestId('edit-buttom')
    waitFor(async () => fireEvent.click(editButtom))

    expect(mockedUsedNavigate).toBeCalledWith(routes.editContact.path(1))
  })
  it('should go to home page', async () => {
    const { findByTestId } = render(
      <TestProviders>
        <ViewContact />
      </TestProviders>
    )
    await waitFor(async () => findByTestId('avatar-container'))

    const button = await findByTestId('back-home-button')

    await waitFor(async () => userEvent.click(button))

    expect(mockedUsedNavigate).toBeCalledWith(routes.home.path)
  })
  it('should remove a contact in the contact list', async () => {
    const { findByTestId } = render(
      <TestProviders>
        <ViewContact />
      </TestProviders>
    )
    await waitFor(async () => findByTestId('avatar-container'))

    const dropdownMenuTrigger = await findByTestId('dropdown-menu-trigger')

    waitFor(() => userEvent.click(dropdownMenuTrigger))

    await waitFor(async () => findByTestId('remove-buttom'))

    const removeButtom = await findByTestId('remove-buttom')
    waitFor(async () => fireEvent.click(removeButtom))

    await waitFor(async () => findByTestId('modal-button-ok'))
    const modalButtonOk = await findByTestId('modal-button-ok')
    waitFor(async () => fireEvent.click(modalButtonOk))

    await waitFor(() =>
      expect(notification).toBeCalledWith(
        'Contato removido com sucesso',
        'success'
      )
    )
    await waitFor(() => expect(deleteContact).toBeCalledWith(1))
  })
})
