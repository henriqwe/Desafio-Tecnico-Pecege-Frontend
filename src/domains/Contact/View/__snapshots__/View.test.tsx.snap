// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewContact should render the contact list correctly 1`] = `
<div>
  <section
    class="sc-grXZZQ jCULEz"
  >
    <div
      class="sc-ftLKQv dNHBDz"
    >
      <div
        class="sc-eDnVMP cJcoqf"
      >
        <div
          data-testid="back-home-button"
        >
          <svg
            aria-hidden="true"
            class="sc-dkjKgF bvwYPB"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.75 19.5L8.25 12l7.5-7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        <h1>
          Contato
        </h1>
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          class="IconButton"
          data-state="closed"
          data-testid="dropdown-menu-trigger"
          id="radix-:r0:"
          type="button"
        >
          <svg
            aria-hidden="true"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 6.75a.75.75 0 110-1.5.75.75 0 010 1.5zM12 12.75a.75.75 0 110-1.5.75.75 0 010 1.5zM12 18.75a.75.75 0 110-1.5.75.75 0 010 1.5z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
      </div>
      <div
        class="sc-jSwlEQ gcFKqT"
      >
        <div
          class="sc-bcPKhP fAwpSg"
          data-testid="avatar-container"
          variant="md"
        >
          L
          G
        </div>
        <p
          class="sc-gKHVLF eNoZjn"
        >
          Leanne Graham
        </p>
        <p
          class="sc-iBAaJG iPRJaf"
        >
          1-770-736-8031 x56442
        </p>
      </div>
    </div>
    <div>
      <div
        class="sc-dIEoRj gozboz"
      >
        <div
          class="sc-hHvkSs kdjjif"
          data-testid="tab-dados"
        >
          <svg
            aria-hidden="true"
            class="sc-pqitP ilFfvI"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.75 6a3.75 3.75 0 11-7.5 0 3.75 3.75 0 017.5 0zM4.501 20.118a7.5 7.5 0 0114.998 0A17.933 17.933 0 0112 21.75c-2.676 0-5.216-.584-7.499-1.632z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          Dados
        </div>
        <div
          class="sc-hHvkSs kYgMLA"
          data-testid="tab-endereco"
        >
          <svg
            aria-hidden="true"
            class="sc-jrkPcm caRRCx"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 10.5a3 3 0 11-6 0 3 3 0 016 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              d="M19.5 10.5c0 7.142-7.5 11.25-7.5 11.25S4.5 17.642 4.5 10.5a7.5 7.5 0 1115 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          Endereço
        </div>
        <div
          class="sc-hHvkSs kdjjif"
          data-testid="tab-empresa"
        >
          <svg
            aria-hidden="true"
            class="sc-kDnyCx hAYOej"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.5 21v-7.5a.75.75 0 01.75-.75h3a.75.75 0 01.75.75V21m-4.5 0H2.36m11.14 0H18m0 0h3.64m-1.39 0V9.349m-16.5 11.65V9.35m0 0a3.001 3.001 0 003.75-.615A2.993 2.993 0 009.75 9.75c.896 0 1.7-.393 2.25-1.016a2.993 2.993 0 002.25 1.016c.896 0 1.7-.393 2.25-1.016a3.001 3.001 0 003.75.614m-16.5 0a3.004 3.004 0 01-.621-4.72L4.318 3.44A1.5 1.5 0 015.378 3h13.243a1.5 1.5 0 011.06.44l1.19 1.189a3 3 0 01-.621 4.72m-13.5 8.65h3.75a.75.75 0 00.75-.75V13.5a.75.75 0 00-.75-.75H6.75a.75.75 0 00-.75.75v3.75c0 .415.336.75.75.75z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          Empresa
        </div>
      </div>
      <div>
        <div
          style="width: 100%; height: 100%; opacity: 0.3027067213852829; transform: translateY(209.18798358441512px) translateZ(0);"
        >
          <section
            class="sc-eLSjS eXINBi"
            data-testid="tab-container"
          >
            <div
              style="width: 100%; display: flex; opacity: 0.00748905250463594; transform: translateX(-297.75328424860925px) translateZ(0);"
            >
              <div
                class="sc-idyqAC hdVvoC"
              >
                <div
                  class="sc-bqOBKd iqZEFD"
                  variant="white"
                >
                  <p
                    class="sc-ipMvLY fHeZYF"
                  >
                    Rua
                  </p>
                  <p
                    class="sc-csmVar ghPXav"
                    data-testid="info-street"
                  >
                    Kulas Light
                  </p>
                </div>
                <div
                  class="sc-bqOBKd gLNhPW"
                  variant="gray"
                >
                  <p
                    class="sc-ipMvLY fHeZYF"
                  >
                    Número
                  </p>
                  <p
                    class="sc-csmVar ghPXav"
                    data-testid="info-suite"
                  >
                    Apt. 556
                  </p>
                </div>
                <div
                  class="sc-bqOBKd iqZEFD"
                  variant="white"
                >
                  <p
                    class="sc-ipMvLY fHeZYF"
                  >
                    Cidade
                  </p>
                  <p
                    class="sc-csmVar ghPXav"
                    data-testid="info-city"
                  >
                    Gwenborough
                  </p>
                </div>
                <div
                  class="sc-bqOBKd gLNhPW"
                  variant="gray"
                >
                  <p
                    class="sc-ipMvLY fHeZYF"
                  >
                    CEP
                  </p>
                  <p
                    class="sc-csmVar ghPXav"
                    data-testid="info-zipcode"
                  >
                    92998-3874
                  </p>
                </div>
              </div>
            </div>
            <div
              style="width: 100%; height: 100%; display: flex; background-color: red; opacity: 0.0030421213605271724; transform: translateX(299.0873635918419px) translateZ(0);"
            >
              <div
                class="sc-fEyylQ kZgmcW"
              />
            </div>
          </section>
        </div>
      </div>
    </div>
  </section>
</div>
`;
